src/NeoSharp.Core/Persistence/IRepositoryOfT.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Linq.Expressions;
+
+namespace NeoSharp.Core.Persistence
+{
+    public interface IRepository<TEntity> 
```

Comment:
I prefer to have readonly version with query methods only as well.

Reply:
What I meant is to have IQueryRepository that contains only Get* methods and ICommandRepository that contains Create/Update/Delete methods __or__ IReadOnlyRepository that contains only Get* methods and IRepository that inherits IReadOnlyRepository and adds Create/Update/Delete methods

Reply:
Agree!